src/shm_main.c

```diff
@@ -768,6 +813,273 @@ sr_shmmain_createunlock(int shm_lock)
     }
 }
 
+/**
+ * @brief Determine whether the connection specified by cid is alive.
+ *
+ * @param[in]  cid Connection ID to check
+ * @param[out] conn_alive Set to non-zero if the connection is alive, 0 otherwise.
+ * @return err_info, NULL on success.
+ */
+sr_error_info_t *
+sr_shmmain_check_conn_lock(sr_cid_t cid, int *conn_alive) {
+    sr_error_info_t *err_info = NULL;
+    struct flock fl;
+    int fd = -1;
+    char *path = NULL;
+    int rc = 0;
+    conn_list_entry *ptr = NULL;
+
+    assert(conn_alive);
+
+    (*conn_alive) = 0;
+
+    /* Connection ID NONE is reserved and never used. */
+    if (SR_CONN_ID_NONE == cid) {
+        return NULL;
+    }
+
+    /* If the connection is owned by this process a check using flock which
+     * requires an open/close would release the lock. Check if the CID is a
+     * connection owned by this process and return status before we do an
+     * open().
+     */
+    if (NULL != conn_head) {
+        if ( (err_info = sr_mlock(conn_list_lock, 1000, __func__))) {
+            return err_info;
+        }
+        for (ptr = conn_head; ptr != NULL; ptr = ptr->_next) {
+            if (cid == ptr->cid) {
+                *conn_alive = 1;
+                sr_munlock(conn_list_lock);
+                return NULL;
+            }
+        }
+        sr_munlock(conn_list_lock);
+    }
+
+    sr_shmmain_get_conn_lockfile_path(cid, &path);
+
+    /* Open the file to test the lock */
+    fd = open(path, O_RDWR);
+    free(path);
+    if (-1 == fd) {
```

Comment:
Generally, please switch these boolean operation operands, so that is is always `variable` operator `value`.